mlflow/store/artifact/mlflow_artifacts_repo.py

```diff
@@ -0,0 +1,12 @@
+from mlflow.store.artifact.http_artifact_repo import HttpArtifactRepository
+from mlflow.tracking._tracking_service.utils import get_tracking_uri
+
+
+class MlflowArtifactsRepository(HttpArtifactRepository):
+    """Scheme wrapper around HttpArtifactRepository for mlflow-artifacts server functionality"""
+
+    def __init__(self, artifact_uri):
+        tracking_uri = get_tracking_uri()
+        resolved_artifacts_uri = artifact_uri.replace("mlflow-artifacts:", f"{tracking_uri}")
```

Comment:
We should only perform this replacement if the URI doesn't specify a host. We also need to handle the case where the URI has one or three slashes after the scheme.

Right now, if the URI specifies a host (e.g. `mlflow-artifacts://myhostname:5000`), the following behavior occurs:
```
In [11]: MlflowArtifactsRepository("mlflow-artifacts://foo:5050/my/artifact/path").artifact_uri
Out[11]: 'http://localhost:5000//foo:5050/my/artifact/path'
```

Similarly, if a hostname-less `mlflow-artifacts` URI is specified with one or three slashes after the scheme, the following behaviors occur:

```
In [14]: mlflow.set_tracking_uri("http://localhost:5000/")

In [15]: MlflowArtifactsRepository("mlflow-artifacts:/my/artifact/path").artifact_uri
Out[15]: 'http://localhost:5000//my/artifact/path'

In [16]: MlflowArtifactsRepository("mlflow-artifacts:///my/artifact/path").artifact_uri
Out[16]: 'http://localhost:5000////my/artifact/path'
```

While the extra slashes in the path shouldn't be harmful, it would be nice to remove them.

To clarify re:hostnames - when a user supplies an artifact URI of the form `mlflow-artifacts://<hostname>/<path>` or `mlflow-artifacts:/<hostname>:<port>/<path>`, we should resolve the URI to `http://<hostname>/api/2.0/mlflow-artifacts/artifacts/<path>` or ``http://<hostname>:<port>/api/2.0/mlflow-artifacts/artifacts/<path>` respectively.

Can we add test cases for each of the above scenarios as well?

Reply:
Some... (not so minor) changes to handle resolving all of the different ways that this uri can be configured. I have a validation check in here for raising an Exception if only a port is passed; I'd really like an opinion on whether this is needed or if it's a silly thing to check for.

Also, should I take those private functions and make them class methods? I don't see those functions being particularly useful outside of this sub class of HttpArtifactRepository. 
@dbczumar @harupy any thoughts / advice on these?

Reply:
> I have a validation check in here for raising an Exception if only a port is passed.

This looks good to me.

> should I take those private functions and make them class methods?

I don't have a strong preference between private functions and class methods. The current code looks ok to me.